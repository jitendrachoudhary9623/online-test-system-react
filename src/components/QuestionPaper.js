import React, { Component } from 'react';
import Timer from "./Timer"
import SubmitButton from "./SubmitButton";
import Question from './Question';
class QuestionPaper extends Component {
    constructor(props) {
        super(props);
        this.state = {"questions":this.props.questions.data,"currentQuestion":0};
        this.timerUpHandler = this.timerUpHandler.bind(this);
        this.score = 0;
        this.currentQuestion=0;
    }
   
    timerUpHandler() {
        let paper = this.state;
        let score = 0;
        paper.questions.map(item => {
            if (item.userAnswer == item.actualAnswer) {
                score++;
            }
        });
        this.score = score;
        this.props.completeTestHandler(score);
    }

    handleAnswerClick = (id, answer) => {
        let questionState = this.state;
        questionState.questions[id - 1].userAnswer = "" + answer;
        for (var counter = 0; counter < questionState.questions[id - 1].Answers.length; counter++) {
            if ('' + (counter + 1) === questionState.questions[id - 1].userAnswer) {
                questionState.questions[id - 1].Answers[counter].marked = true;
            }
            else {
                questionState.questions[id - 1].Answers[counter].marked = false;
            }
        }
        this.setState({ questionState });
    }

    nextButtonHandler(){
        let tempState=this.state;
        if(tempState.questions.length-1>this.currentQuestion){
            this.currentQuestion++;
            this.setState({tempState});
        }

    }

    previousButtonHandler=()=>{
        if(this.currentQuestion>0){
            this.currentQuestion--;
            let tempState=this.state;
            this.setState(tempState);
        }
    }
    render() {

        return (<div className="row">
            <div className="col s8">
                <Question key={'b' + this.state.questions[this.currentQuestion].id} item={this.state.questions[this.currentQuestion]} id={'a' + this.state.questions[this.currentQuestion].id} handleAnswerClick={this.handleAnswerClick}/>
            </div>
            <div className="col s3 navigationStyle">
            <Timer seconds={this.props.seconds} timerUpHandler={this.timerUpHandler} />
            <button onClick={this.nextButtonHandler.bind(this)} className="row col s12 btn grey lighten-5 buttonStyle" style={{color:"black"}}>Next </button>
            <button onClick={this.previousButtonHandler.bind(this)} className="row col s12 btn grey lighten-5 buttonStyle" style={{color:"black"}}>Previous </button>
            <SubmitButton timerUpHandler={this.timerUpHandler} />

            </div>
           

        </div>);
    }

};

export default QuestionPaper;