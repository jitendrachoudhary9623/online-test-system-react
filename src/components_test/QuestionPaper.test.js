import React, { Component } from 'react';
import QuestionPaper from "../components/client/QuestionPaper.component"
import { shallow } from 'enzyme'
import data from "../datasource/mockData"
describe('UI Test for Question Paper Button', () => {
    let questions = {"data":data}
    let snapshot = shallow(
        <QuestionPaper questions={questions} seconds={5}/>
      )
    it('Question Paper to be defined', () => {
        expect(QuestionPaper).toBeDefined();
    });

    it('Question Paper  should render correctly', () => {
        expect(snapshot).toMatchSnapshot();
      });
})