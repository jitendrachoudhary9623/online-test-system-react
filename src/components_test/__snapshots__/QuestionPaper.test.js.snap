exports[`UI Test for Question Paper Button Question Paper  should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <QuestionPaper
    questions={
        Object {
            "data": Array [
              Object {
                "Answers": Array [
                  Object {
                    "marked": false,
                    "option": "6.54 sec",
                  },
                  Object {
                    "marked": false,
                    "option": "44.32 sec",
                  },
                  Object {
                    "marked": false,
                    "option": "55 sec",
                  },
                  Object {
                    "marked": false,
                    "option": "30.2 sec",
                  },
                ],
                "Question": "A train 120 meters long is running with a speed of 60 km/hr. In what time will it pass a boy who is running at 6 km/hr in the direction opposite to that in which the train is going?",
                "actualAnswer": "1",
                "id": 1,
                "type": "objective",
                "userAnswer": "",
              },
              Object {
                "Answers": Array [
                  Object {
                    "marked": false,
                    "option": "8 km",
                  },
                  Object {
                    "marked": false,
                    "option": "9 km",
                  },
                  Object {
                    "marked": false,
                    "option": "6 km",
                  },
                  Object {
                    "marked": false,
                    "option": "7.5 km",
                  },
                ],
                "Question": "Devesh can cover a certain distance in 1 hour 24 minutes by covering two-third of the distance at 6 km/hour and the rest at 7 km/hr. Calculate total distance.",
                "actualAnswer": "3",
                "id": 2,
                "type": "objective",
                "userAnswer": "",
              },
              Object {
                "Answers": Array [
                  Object {
                    "marked": false,
                    "option": "6.54 sec",
                  },
                  Object {
                    "marked": false,
                    "option": "44.32 sec",
                  },
                  Object {
                    "marked": false,
                    "option": "55 sec",
                  },
                  Object {
                    "marked": false,
                    "option": "30.2 sec",
                  },
                ],
                "Question": "A train 120 meters long is running with a speed of 60 km/hr. In what time will it pass a boy who is running at 6 km/hr in the direction opposite to that in which the train is going?",
                "actualAnswer": "1",
                "id": 3,
                "type": "objective",
                "userAnswer": "",
              },
              Object {
                "Answers": Array [
                  Object {
                    "marked": false,
                    "option": "8 km",
                  },
                  Object {
                    "marked": false,
                    "option": "9 km",
                  },
                  Object {
                    "marked": false,
                    "option": "6 km",
                  },
                  Object {
                    "marked": false,
                    "option": "7.5 km",
                  },
                ],
                "Question": "Devesh can cover a certain distance in 1 hour 24 minutes by covering two-third of the distance at 6 km/hour and the rest at 7 km/hr. Calculate total distance.",
                "actualAnswer": "3",
                "id": 4,
                "type": "objective",
                "userAnswer": "",
              },
            ],
          }
    }
    seconds={5} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col s8">
          <Question
                    handleAnswerClick={[Function]}
                    id="a1"
                    item={
                              Object {
                                        "Answers": Array [
                                          Object {
                                            "marked": false,
                                            "option": "6.54 sec",
                                          },
                                          Object {
                                            "marked": false,
                                            "option": "44.32 sec",
                                          },
                                          Object {
                                            "marked": false,
                                            "option": "55 sec",
                                          },
                                          Object {
                                            "marked": false,
                                            "option": "30.2 sec",
                                          },
                                        ],
                                        "Question": "A train 120 meters long is running with a speed of 60 km/hr. In what time will it pass a boy who is running at 6 km/hr in the direction opposite to that in which the train is going?",
                                        "actualAnswer": "1",
                                        "id": 1,
                                        "type": "objective",
                                        "userAnswer": "",
                                      }
                    } />
</div>,
        <div
          className="col s3 navigationStyle">
          <Timer
                    seconds={5}
                    timerUpHandler={[Function]} />
          <button
                    className="row col s12 btn grey lighten-5 buttonStyle"
                    onClick={[Function]}
                    style={
                              Object {
                                        "color": "black",
                                      }
                    }>
                    Next 
          </button>
          <button
                    className="row col s12 btn grey lighten-5 buttonStyle"
                    onClick={[Function]}
                    style={
                              Object {
                                        "color": "black",
                                      }
                    }>
                    Previous 
          </button>
          <SubmitButton
                    timerUpHandler={[Function]} />
</div>,
      ],
      "className": "row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Question
            handleAnswerClick={[Function]}
            id="a1"
            item={
                        Object {
                                    "Answers": Array [
                                      Object {
                                        "marked": false,
                                        "option": "6.54 sec",
                                      },
                                      Object {
                                        "marked": false,
                                        "option": "44.32 sec",
                                      },
                                      Object {
                                        "marked": false,
                                        "option": "55 sec",
                                      },
                                      Object {
                                        "marked": false,
                                        "option": "30.2 sec",
                                      },
                                    ],
                                    "Question": "A train 120 meters long is running with a speed of 60 km/hr. In what time will it pass a boy who is running at 6 km/hr in the direction opposite to that in which the train is going?",
                                    "actualAnswer": "1",
                                    "id": 1,
                                    "type": "objective",
                                    "userAnswer": "",
                                  }
            } />,
          "className": "col s8",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": "b1",
          "nodeType": "class",
          "props": Object {
            "handleAnswerClick": [Function],
            "id": "a1",
            "item": Object {
              "Answers": Array [
                Object {
                  "marked": false,
                  "option": "6.54 sec",
                },
                Object {
                  "marked": false,
                  "option": "44.32 sec",
                },
                Object {
                  "marked": false,
                  "option": "55 sec",
                },
                Object {
                  "marked": false,
                  "option": "30.2 sec",
                },
              ],
              "Question": "A train 120 meters long is running with a speed of 60 km/hr. In what time will it pass a boy who is running at 6 km/hr in the direction opposite to that in which the train is going?",
              "actualAnswer": "1",
              "id": 1,
              "type": "objective",
              "userAnswer": "",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Timer
              seconds={5}
              timerUpHandler={[Function]} />,
            <button
              className="row col s12 btn grey lighten-5 buttonStyle"
              onClick={[Function]}
              style={
                            Object {
                                          "color": "black",
                                        }
              }>
              Next 
</button>,
            <button
              className="row col s12 btn grey lighten-5 buttonStyle"
              onClick={[Function]}
              style={
                            Object {
                                          "color": "black",
                                        }
              }>
              Previous 
</button>,
            <SubmitButton
              timerUpHandler={[Function]} />,
          ],
          "className": "col s3 navigationStyle",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "seconds": 5,
              "timerUpHandler": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Next ",
              "className": "row col s12 btn grey lighten-5 buttonStyle",
              "onClick": [Function],
              "style": Object {
                "color": "black",
              },
            },
            "ref": null,
            "rendered": "Next ",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Previous ",
              "className": "row col s12 btn grey lighten-5 buttonStyle",
              "onClick": [Function],
              "style": Object {
                "color": "black",
              },
            },
            "ref": null,
            "rendered": "Previous ",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "timerUpHandler": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col s8">
            <Question
                        handleAnswerClick={[Function]}
                        id="a1"
                        item={
                                    Object {
                                                "Answers": Array [
                                                  Object {
                                                    "marked": false,
                                                    "option": "6.54 sec",
                                                  },
                                                  Object {
                                                    "marked": false,
                                                    "option": "44.32 sec",
                                                  },
                                                  Object {
                                                    "marked": false,
                                                    "option": "55 sec",
                                                  },
                                                  Object {
                                                    "marked": false,
                                                    "option": "30.2 sec",
                                                  },
                                                ],
                                                "Question": "A train 120 meters long is running with a speed of 60 km/hr. In what time will it pass a boy who is running at 6 km/hr in the direction opposite to that in which the train is going?",
                                                "actualAnswer": "1",
                                                "id": 1,
                                                "type": "objective",
                                                "userAnswer": "",
                                              }
                        } />
</div>,
          <div
            className="col s3 navigationStyle">
            <Timer
                        seconds={5}
                        timerUpHandler={[Function]} />
            <button
                        className="row col s12 btn grey lighten-5 buttonStyle"
                        onClick={[Function]}
                        style={
                                    Object {
                                                "color": "black",
                                              }
                        }>
                        Next 
            </button>
            <button
                        className="row col s12 btn grey lighten-5 buttonStyle"
                        onClick={[Function]}
                        style={
                                    Object {
                                                "color": "black",
                                              }
                        }>
                        Previous 
            </button>
            <SubmitButton
                        timerUpHandler={[Function]} />
</div>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Question
              handleAnswerClick={[Function]}
              id="a1"
              item={
                            Object {
                                          "Answers": Array [
                                            Object {
                                              "marked": false,
                                              "option": "6.54 sec",
                                            },
                                            Object {
                                              "marked": false,
                                              "option": "44.32 sec",
                                            },
                                            Object {
                                              "marked": false,
                                              "option": "55 sec",
                                            },
                                            Object {
                                              "marked": false,
                                              "option": "30.2 sec",
                                            },
                                          ],
                                          "Question": "A train 120 meters long is running with a speed of 60 km/hr. In what time will it pass a boy who is running at 6 km/hr in the direction opposite to that in which the train is going?",
                                          "actualAnswer": "1",
                                          "id": 1,
                                          "type": "objective",
                                          "userAnswer": "",
                                        }
              } />,
            "className": "col s8",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": "b1",
            "nodeType": "class",
            "props": Object {
              "handleAnswerClick": [Function],
              "id": "a1",
              "item": Object {
                "Answers": Array [
                  Object {
                    "marked": false,
                    "option": "6.54 sec",
                  },
                  Object {
                    "marked": false,
                    "option": "44.32 sec",
                  },
                  Object {
                    "marked": false,
                    "option": "55 sec",
                  },
                  Object {
                    "marked": false,
                    "option": "30.2 sec",
                  },
                ],
                "Question": "A train 120 meters long is running with a speed of 60 km/hr. In what time will it pass a boy who is running at 6 km/hr in the direction opposite to that in which the train is going?",
                "actualAnswer": "1",
                "id": 1,
                "type": "objective",
                "userAnswer": "",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Timer
                seconds={5}
                timerUpHandler={[Function]} />,
              <button
                className="row col s12 btn grey lighten-5 buttonStyle"
                onClick={[Function]}
                style={
                                Object {
                                                "color": "black",
                                              }
                }>
                Next 
</button>,
              <button
                className="row col s12 btn grey lighten-5 buttonStyle"
                onClick={[Function]}
                style={
                                Object {
                                                "color": "black",
                                              }
                }>
                Previous 
</button>,
              <SubmitButton
                timerUpHandler={[Function]} />,
            ],
            "className": "col s3 navigationStyle",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "seconds": 5,
                "timerUpHandler": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Next ",
                "className": "row col s12 btn grey lighten-5 buttonStyle",
                "onClick": [Function],
                "style": Object {
                  "color": "black",
                },
              },
              "ref": null,
              "rendered": "Next ",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Previous ",
                "className": "row col s12 btn grey lighten-5 buttonStyle",
                "onClick": [Function],
                "style": Object {
                  "color": "black",
                },
              },
              "ref": null,
              "rendered": "Previous ",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "timerUpHandler": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
